### DISCUSSION TITLE
WOE encoding for continuous variable

### DISCUSSION THREAD

**Discussion by jnsofini** [guillermo-navas-palencia/optbinning#305] [Q&A] [answered]
I read the FICO article you cited on page 13 of Optimal Binning article from 2022. The equation for WOE and IV (for a continuous target) is a good heuristic there, however, I found this [article from the department of statistic from the University of Maryland](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwiH2qSfsfCEAxUBkokEHaTyDbAQFnoECBMQAQ&url=https%3A%2F%2Fwww.lexjansen.com%2Fsesug%2F2014%2FSD-20.pdf&usg=AOvVaw2d2f0hbwHYm8bTLT6duii3&opi=89978449)  which gives an alternate definition that is very close to the original WOE and IV for binary case. Then even run some benchmarks with the alternative definition for the binary target and they show amazing match. 

I find that there are significant scorecard points for the binary and continuous case that is implemented in optbinning now. Could that be solved by this other encoding in this article?

**Comment by ericlecoutre** [comment, marked as answer]
Hi,

I confirm: I have implemented alternate method described here in a R package (basically slightly changing just normalizing target [0;1] and empirical tests show good behaviors.
I didn't propose a PR because:
- I am not Python  fluent
- I fear it might have some impacts on empirical measure that tries to compromize between various parameters (IV, nombre bins...)  

I would really be happy to see such change or additional option as parameter implemented.

**Reply by jnsofini** [reply]
@guillermo-navas-palencia would this be possible, I think the optimization would be similar to the binary case if this form of IV is implemented.

**Comment by guillermo-navas-palencia** [comment]
Hi @jnsofini @ericlecoutre.

Thanks for opening the discussion.

I came across this paper about an alternative WoE definition several times throughout the last few years.  In my opinion, this definition is not sufficiently generic, although it might be useful for cases like the one presented (sales per bin). The current WoE for continuous target, similar to the implementation for binary target, informs on the difference to the average target, which I consider more valuable than metrics based on the relative bin size.

Then, at this moment, I do not consider adding alternative metric definitions to the library.

**Reply by jnsofini** [reply]
Thank you! The concept of WoE for continuous was vague to me. Glad I found it in Optbinning library. Great work!
